mmedit/models/restorers/liif.py

```diff
@@ -0,0 +1,411 @@
+
+
+@MODELS.register_module()
+class LIIF(BasicRestorer):
+    """LIIF model for single image super-resolution.
+
+    Paper: Learning Continuous Image Representation with
+           Local Implicit Image Function
+    Ref repo: https://github.com/yinboc/liif.git
+
+    Args:
+        generator (dict): Config for the generator.
+        imnet (dict): Config for the imnet.
+        local_ensemble (bool): Whether to use local ensemble.
+            Default: True.
+        feat_unfold (bool): Whether to use feat unfold. Default: True.
+        cell_decode (bool): Whether to use cell decode. Default: True.
+        data_mean (list[float]): Data mean.
+            Default: [0.5].
+        data_std (list[float]): Data std.
+            Default: [0.5].
+        eval_bsize (int): Size of batched predict. Default: None.
+        pixel_loss (dict): Config for the pixel loss. Default: None.
+        train_cfg (dict): Config for train. Default: None.
+        test_cfg (dict): Config for testing. Default: None.
+        pretrained (str): Path for pretrained model. Default: None.
+    """
+
+    def __init__(self,
+                 generator,
+                 imnet,
+                 local_ensemble=True,
+                 feat_unfold=True,
+                 cell_decode=True,
+                 data_mean=[0.5],
+                 data_std=[0.5],
+                 eval_bsize=None,
+                 pixel_loss=None,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+
+        super().__init__(generator, pixel_loss, train_cfg, test_cfg,
+                         pretrained)
+
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        self.local_ensemble = local_ensemble
+        self.feat_unfold = feat_unfold
+        self.cell_decode = cell_decode
+        self.eval_bsize = eval_bsize
+        self.feat = None
+
+        # norm
+        data_mean = torch.FloatTensor(data_mean)
+        data_std = torch.FloatTensor(data_std)
+        self.lq_mean = data_mean.view(1, -1, 1, 1)
+        self.lq_std = data_std.view(1, -1, 1, 1)
+        self.gt_mean = data_mean.view(1, 1, -1)
+        self.gt_std = data_std.view(1, 1, -1)
+
+        # model
+        generator_model = build_backbone(generator)
+        generator_list = list(generator_model.children())
+        self.head = generator_list[0]
+        self.encoder = nn.Sequential(*generator_list[1:-2])
+        imnet_in_dim = generator_model.mid_channels
+        if self.feat_unfold:
+            imnet_in_dim *= 9
+        imnet_in_dim += 2  # attach coord
+        if self.cell_decode:
+            imnet_in_dim += 2
+        imnet['in_dim'] = imnet_in_dim
+        self.imnet = build_component(imnet)
+
+        # support fp16
+        self.fp16_enabled = False
+
+        # loss
+        self.pixel_loss = build_loss(pixel_loss) if pixel_loss else None
+
+        self.step_counter = 0  # counting training steps
```

Comment:
What if the model is resumed

Reply:
This parameter (`step_counter`) is actually not used and has been deleted.